osp/core/ontology/namespace.py

```diff
@@ -113,86 +130,233 @@ def __getitem__(self, label):
+
+    def get_from_suffix(self, suffix, case_sensitive=False):
+        """Get an ontology entity from its namespace suffix.
+
+        Args:
+            suffix (str): Suffix of the ontology entity.
+            case_sensitive (bool): Whether to search also for the same suffix
+                                   with different capitalization. By default,
+                                   such a search is performed.
+        """
+        iri = rdflib.URIRef(str(self._iri) + suffix)
         try:
-            return self._do_get(name, _case_sensitive, _force_by_iri)
+            return self.get_from_iri(iri, _name=suffix)
         except KeyError as e:
-            if name and name.startswith("INVERSE_OF_"):
-                return self._do_get(name[11:], _case_sensitive,
-                                    _force_by_iri).inverse
+            if not case_sensitive:
+                return self._get_case_insensitive(suffix,
+                                                  self.get_from_suffix)
             raise e
 
-    def _do_get(self, name, _case_sensitive, _force_by_iri):
+    def _get_from_label(self, label, lang=None, case_sensitive=False):
+        """Get an ontology entity from the registry by label.
+
+        Args:
+            label (str): The label of the ontology entity.
+
+        Raises:
+            KeyError: Unknown label.
+
+        Returns:
+            OntologyEntity: The ontology entity.
+        """
+        results = []
+        for iri in self._get_namespace_subjects():
+            entity_labels = self._get_labels_for_iri(iri, lang=lang)
+            if case_sensitive is False:
+                entity_labels = (label.lower() for label in entity_labels)
+                comp_label = label.lower()
+            else:
+                comp_label = label
+            if comp_label in entity_labels:
+                _name = label if self._reference_by_label else None
+                results.append(self.get_from_iri(iri, _name=_name))
+        else:
+            if len(results) == 0:
+                raise KeyError("No element with label %s in namespace %s"
+                               % (label, self))
+            elif len(results) >= 2:
+                element_suffixes = (r._iri_suffix for r in results)
+                raise KeyError(f"There are multiple elements "
+                               f"({', '.join(element_suffixes)}) with label"
+                               f" {label} for namespace {self}."
+                               f"\n"
+                               f"Please refer to a specific element of the "
+                               f"list by calling get_from_iri(IRI) for "
+                               f"namespace {self} for one of the following "
+                               f"IRIs: " + "{iris}."
+                               .format(iris=', '.join(entity.iri for entity in
+                                                      results)))
+            else:
```

Comment:
I would also remove this else. Both cases before result in an exception, so you can simply return results[0] outside of the else.

Reply:
Also removed in future commit.